test/native-tests/test-weakref/sample.cpp

```diff
@@ -0,0 +1,71 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "ChakraCore.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include <string>
+#include <cstring>
+
+#define FAIL_CHECK(cmd)                     \
+    do                                      \
+    {                                       \
+        JsErrorCode errCode = cmd;          \
+        if (errCode != JsNoError)           \
+        {                                   \
+            printf("Error %d at '%s'\n",    \
+                errCode, #cmd);             \
+            return 1;                       \
+        }                                   \
+    } while(0)
+
+using namespace std;
+
+int main()
+{
+    JsRuntimeHandle runtime;
+    JsContextRef context;
+    unsigned currentSourceContext = 0;
+
+    // Create a runtime.
+    JsCreateRuntime(JsRuntimeAttributeNone, nullptr, &runtime);
+
+    // Create an execution context.
+    JsCreateContext(runtime, &context);
+
+    // Now set the current execution context.
+    JsSetCurrentContext(context);
+
+    JsValueRef valueRef;
+    FAIL_CHECK(JsCreateString("sample", strlen("sample"), &valueRef));
+
+    JsWeakRef weakRef;
+    FAIL_CHECK(JsCreateWeakReference(valueRef, &weakRef));
+
+    FAIL_CHECK(JsGetWeakReferenceValue(weakRef, &valueRef));
```

Comment:
Would it make sense to fetch the value to a new reference, and check that `valueRef == weakRefValue`?